dali/pipeline/data/tensor_vector.h

```diff
@@ -123,6 +123,12 @@ class DLL_PUBLIC TensorVector {
 
   DLL_PUBLIC void Resize(const TensorListShape<> &new_shape, const TypeInfo &new_type);
 
+  /**
+   * Change the number of tensors in TensorVector.
+   * @param new_size
+   */
+  void SetSize(int new_size);
```

Comment:
Maybe it should just accept an unsigned value and prevent from the negative one?